### PR TITLE
Frontend/Feat: #BKN-3043 Added the special Zone(Cochin Airport) featu…

### PR DESCRIPTION
**Pull request by guptamanav422**
Added the special Zone Feature in Yatri App!

## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added the special Zone feature in Yatri APP (Cochin International Airport)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 1]
```diff
@@ -1995,7 +1995,7 @@ estimatesFlow estimatedQuotes state = do
         Nothing -> 0
       showRateCardIcon = if (null estimateFareBreakup) then false else true
       zoneType = getSpecialTag $ if isJust (estimatedVarient !! 0) then (fromMaybe dummyEstimateEntity (estimatedVarient !! 0)) ^. _specialLocationTag else Nothing
-  if ((not (null estimatedVarient)) && (isLocalStageOn FindingEstimate)) then do
+  if not (null estimatedVarient) then do
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:1998]: why this Local stage check is removed?

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 2]
```diff
@@ -899,7 +899,7 @@ rideRequestFlowView push state =
         ]
         [ PrestoAnim.animationSet [ fadeIn true ]
             $ if (state.props.currentStage == SettingPrice) then
-                if (state.props.isSpecialZone && (getMerchant FunctionCall == JATRISAATHI))  || ((getMerchant FunctionCall) /= NAMMAYATRI) then
+                if (state.props.isSpecialZone && ((getValueFromConfig "specialLocationView") == "true"))  || ((getMerchant FunctionCall) /= NAMMAYATRI) then
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:902]: remove extra brackets

Comment (guptamanav422) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs]: Resolved

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs (hunk 1 above)
Comment (guptamanav422) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs]: Resolved, there is no need of removing this check

File: Frontend/ui-customer/src/Flow.purs [hunk 3]
```diff
@@ -918,7 +918,7 @@ homeScreenFlow = do
           if state.homeScreen.props.sourceLat/=0.0 && state.homeScreen.props.sourceLong /= 0.0 then do
             (ServiceabilityRes sourceServiceabilityResp) <- Remote.originServiceabilityBT (Remote.makeServiceabilityReq state.homeScreen.props.sourceLat state.homeScreen.props.sourceLong)
             if (sourceServiceabilityResp.serviceable ) then do
-              if (isJust sourceServiceabilityResp.geoJson && (getMerchant FunctionCall) == JATRISAATHI ) then do
+              if (isJust sourceServiceabilityResp.geoJson && (MU.getValueFromConfig "specialLocationView") == "true") then do
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Flow.purs:921]: @guptamanav422 can we remove the brackets from here. As it is not required.

File: Frontend/ui-customer/src/Flow.purs [hunk 4]
```diff
@@ -1201,7 +1201,7 @@ rideSearchFlow flowType = do
     then do
       case ((not finalState.props.isSpecialZone) && finalState.props.sourceSelectedOnMap) of
         false -> do
-          _ <- pure $ locateOnMap false finalState.props.sourceLat finalState.props.sourceLong (if (getMerchant FunctionCall == JATRISAATHI) then finalState.data.polygonCoordinates else "")  (if (getMerchant FunctionCall == JATRISAATHI) then finalState.data.nearByPickUpPoints else [])
+          _ <- pure $ locateOnMap false finalState.props.sourceLat finalState.props.sourceLong finalState.data.polygonCoordinates finalState.data.nearByPickUpPoints
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Flow.purs:1204]: @guptamanav422 is this required for namma Yatri also?

**Pull request closed without merging**